front/src/reducers/infra/selectors.ts

```diff
@@ -0,0 +1,15 @@
+import { makeSubSelector } from 'utils/selectors';
+
+import type { RootState, OsrdSlicesType } from 'reducers';
+import type { InfraStateType } from 'reducers/infra/index';
+
+const buildInfraStateSelectors = <infraState extends InfraStateType>(slice: OsrdSlicesType) => {
+  const getInfra = (state: RootState) => state[slice.name] as infraState;
+  const makeInfraSelector = makeSubSelector<infraState>(getInfra);
+  return {
+    getInfraID: makeInfraSelector('infraID'),
+    getFeatureInfoClick: makeInfraSelector('featureInfoClick'),
+  };
+};
```

Comment:
You can simplify it like this
```suggestion
const buildInfraStateSelectors = (slice: OsrdSlicesType) => ({
  getInfraID: (state: RootState) => state[slice.name].infraID,
});
```

Reply:
Done